de.fu_berlin.inf.dpp.server/src/de/fu_berlin/inf/dpp/server/console/ServerConsole.java

```diff
@@ -0,0 +1,58 @@
+package de.fu_berlin.inf.dpp.server.console;
+
+import java.io.InputStream;
+import java.io.PrintStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+public class ServerConsole implements Runnable {
+    private Scanner input;
+    private PrintStream output;
+    private List<ConsoleCommand> commands;
+
+    public ServerConsole(InputStream in, PrintStream out) {
+        input = new Scanner(in);
+        output = out;
+        commands = new ArrayList<>();
+    }
+
+    public void registerCommand(ConsoleCommand command) {
+        commands.add(command);
+    }
+
+    @Override
+    public void run() {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Refactor this method to reduce its Cognitive Complexity from 22 to the 15 allowed. ([squid:S3776](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS3776))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=143&issues=AWVmdMM4elKyZq-pzZKs&open=AWVmdMM4elKyZq-pzZKs)

[ISSUE_KEY]: AWVmdMM4elKyZq-pzZKs